File: typescript/tmp6/main.ts
function main() {
    const nums = [1, 2, 3, 4];
    const target = 7;
    let resp = run(nums, target)
    console.log(resp)
}

function two_sum(nums: number[], target: number): number {
    if (nums.length) < 2 return [];

    const seen = new Map<number, number>();

    for (let i = 0; i < nums.length; i++) {
        const complement = target - seen[i];

        if (seen.has(complement)) {

        }
    }
}

main()

File: cpp/tmp6/main.cpp
#include <iostream>
#include <vector>
#include <unordered_map>

std::pair<int, int> run(const std::vector<int>& nums, int target) {
    if (nums.size() < 2) {
        return {-1, -1};
    }

    std::unordered_map<int, int> seen;

    for (int idx = 0; idx < nums.size(); idx++) {
        int num = nums[idx];
        int complement = target - num;

        if (seen.find(complement) != seen.end()) {
            int compIdx = seen[complement]; 
            return {compIdx, idx};
        }

        seen[num] = idx;
    }

    return {-1, -1};
}

int main() {
    std::vector<int> nums = {1, 2, 3, 4};
    int target = 7;

    std::pair<int, int> result = run(nums, target);

    if (result.first != -1) {
        std::cout << result.first << result.second << std::endl;
    } else {
        std::cout << "False" << std::endl;
    }

    return 0;
}
